core/gui/src/app/workspace/service/workflow-result-export/workflow-result-export.service.ts

```diff
@@ -60,34 +69,284 @@ export class WorkflowResultExportService {
       this.workflowActionService.getJointGraphWrapper().getJointOperatorHighlightStream(),
       this.workflowActionService.getJointGraphWrapper().getJointOperatorUnhighlightStream()
     ).subscribe(() => {
-      // check if there are any results to export on highlighted operators (either paginated or snapshot)
-      this.hasResultToExportOnHighlightedOperators =
-        isNotInExecution(this.executeWorkflowService.getExecutionState().state) &&
-        this.workflowActionService
-          .getJointGraphWrapper()
-          .getCurrentHighlightedOperatorIDs()
-          .filter(
-            operatorId =>
-              this.workflowResultService.hasAnyResult(operatorId) ||
-              this.workflowResultService.getResultService(operatorId)?.getCurrentResultSnapshot() !== undefined
-          ).length > 0;
-
-      // check if there are any results to export on all operators (either paginated or snapshot)
-      let staticHasResultToExportOnAllOperators =
-        isNotInExecution(this.executeWorkflowService.getExecutionState().state) &&
-        this.workflowActionService
-          .getTexeraGraph()
-          .getAllOperators()
-          .map(operator => operator.operatorID)
-          .filter(
-            operatorId =>
-              this.workflowResultService.hasAnyResult(operatorId) ||
-              this.workflowResultService.getResultService(operatorId)?.getCurrentResultSnapshot() !== undefined
-          ).length > 0;
-
-      // Notify subscribers of changes
-      this.hasResultToExportOnAllOperators.next(staticHasResultToExportOnAllOperators);
+      this.updateExportAvailabilityFlags();
+    });
+  }
+
+  /**
+   * Computes restriction analysis on-demand by fetching dataset metadata and analyzing workflow graph.
+   *
+   * Fetches all accessible datasets and their permissions, then:
+   * - Builds datasetDownloadableMap: tracks which datasets are downloadable
+   * - Builds datasetLabelMap: stores human-readable dataset labels
+   * - Performs restriction analysis to identify operators blocked by dataset access controls
+   *
+   * A dataset is considered downloadable if either:
+   * - The dataset's isDownloadable flag is true, OR
+   * - The current user is the dataset owner
+   *
+   * @returns Observable that emits the restriction analysis result
+   */
+  public computeRestrictionAnalysis(): Observable<RestrictionAnalysisResult> {
```

Comment:
I saw on the backend you add an endpoint to calculate the same thing as here. Can you use that endpoint to replace the current frontend implementation?